src/main/java/com/arpnetworking/commons/builder/ValidationProcessor.java

```diff
@@ -0,0 +1,288 @@
+import net.sf.oval.constraint.MaxSize;
+import net.sf.oval.constraint.MemberOf;
+import net.sf.oval.constraint.Min;
+import net.sf.oval.constraint.MinLength;
+import net.sf.oval.constraint.MinSize;
+import net.sf.oval.constraint.NotBlank;
+import net.sf.oval.constraint.NotEmpty;
+import net.sf.oval.constraint.NotEqual;
+import net.sf.oval.constraint.NotMatchPattern;
+import net.sf.oval.constraint.NotMemberOf;
+import net.sf.oval.constraint.NotNegative;
+import net.sf.oval.constraint.NotNull;
+import net.sf.oval.constraint.Past;
+import net.sf.oval.constraint.Range;
+import net.sf.oval.constraint.Size;
+
+import java.lang.annotation.Annotation;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+
+/**
+ * Implementation of {@link ClassProcessor} for adding code to {@link OvalBuilder}
+ * implementations to validate constraints. This replaces the reflection based
+ * validation in the OVal framework but uses the same check classes.
+ *
+ * Dependencies:
+ * <ul>
+ *     <li>com.arpnetworking.commons:javassist-maven-plugin (as plugin only)</li>
+ *     <li>com.arpnetworking.commons:javassist-maven-core</li>
+ *     <li>net.sf.oval:oval</li>
+ *     <li>com.google.guava:guava</li>
+ * </ul>
+ *
+ * @author Ville Koskela (ville dot koskela at inscopemetrics dot com)
+ */
+public final class ValidationProcessor implements ClassProcessor {
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean accept(final CtClass ctClass) {
+        // Reject processing of OvalBuilder itself
+        if (OVAL_BUILDER_CLASS.equals(ctClass.getName())) {
+            return false;
+        }
+        // Reject any classes annotated with SkipValidationProcessor
+        try {
+            for (final Object object : ctClass.getAnnotations()) {
+                final Annotation annotation = (Annotation) object;
+                if (SKIP_VALIDATION_TRANSFORM_CLASS.equals(annotation.annotationType().getName())) {
+                    return false;
+                }
+            }
+        } catch (final ClassNotFoundException e) {
+            throw new RuntimeException(String.format("Unable to evaluate class %s", ctClass.getName()), e);
+        }
+        // Accept any classes which descend from OvalBuilder
+        CtClass parent = ctClass;
+        while (parent != null) {
+            if (OVAL_BUILDER_CLASS.equals(parent.getName())) {
+                return true;
+            }
+            try {
+                parent = parent.getSuperclass();
+            } catch (final NotFoundException e) {
+                throw new RuntimeException(String.format("Unable to evaluate class %s", ctClass.getName()), e);
+            }
+        }
+        // All other classes are rejected
+        return false;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @SuppressFBWarnings("REC_CATCH_EXCEPTION")
+    @Override
+    public void process(final CtClass ctClass) {
+        try {
+            final String collectSuperClassViolations;
```

Comment:
Some explanation as to why you're collecting (a chain?) of methods to find violations is in order.  Also, what happens if the super class wasn't weaved?  Note: make sure you have a test for this situation somewhere.